prov/rxd/src/rxd_cq.c

```diff
@@ -917,10 +974,13 @@ static void rxd_handle_data(struct rxd_ep *ep, struct rxd_pkt_entry *pkt_entry)
 		rxd_remove_rx_pkt(ep, pkt_entry);
 		dlist_insert_order(&ep->peers[pkt->base_hdr.peer].buf_pkts,
 				   &rxd_comp_pkt_seq_no, &pkt_entry->d_entry);
-		ep->peers[pkt->base_hdr.peer].rx_seq_no++;
+		return;
 	} else {
```

Comment:
nit: you can remove the else now

Reply:
Can I?

Reply:
But the initial if doesn't always return?

Reply:
Ah - github was hiding the else if statement from the view